tensorflow/python/ops/array_grad.py

```diff
@@ -504,6 +504,8 @@ def _ExtractImagePatchesGrad(op, grad):
   batch_size, rows_in, cols_in, channels = [
     dim.value for dim in op.inputs[0].get_shape()
   ]
+  batch_size = array_ops.shape(op.inputs[0])[0]
+
```

Comment:
To be consistent, you should also use array_ops.shape() to extract rows_in, cols_in, and channels